### ISSUE TITLE
RKE cluster can not be accessed via UI if cni is set to none

### ISSUE THREAD

**Issue by ranchersvet** [rancher/rancher#30427]
**What kind of request is this (question/bug/enhancement/feature request):** bug

**Steps to reproduce (least amount of steps as possible):**
Build a custom cluster using Rancher v2.5.3. Edit cluster.yml file and set cni to none:
```
network:
    plugin: none
```

**Result:**
If you try to download kubeconfig file or to launch kubectl they will be grayed out.
Cluster is still accessible via rancher cli and third-party CNI can be installed.

**Other details that may be helpful:**
When using version 2.4.x buttons launch kubectl and download kubeconfig file are available and third-party CNI can be installed via GUI.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.5.3
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Custom

gz#13843
gz#15513

**Comment by westlywright** [comment]
@ranchersvet I had a cursory look at this and wanted to share my findings. This doesn't seem to be related to the UI. The code that enables those buttons uses a flag on the cluster model in the UI, `isReady`. This flag checks the conditions of the current cluster and looks for a condition with `{ type: Ready, status: True}`. In the 2.4 cluster that @luthermonson provided the cluster without a CNI provider had a ready status. On an equivalent cluster in 2.5 (also provided by @luthermonson ) the cluster did not have a `True` ready status. I suspect the backend changed how we set Ready.

**Comment by jiaqiluo** [comment]
Here is the behavior I observed on rancher:`v2.4.13`

Steps:
- run rancher single-install
- add a custom cluster, edit the cluster as YAML and replace the default network settings with the following 
```
network:
    plugin: none
```
- register one node with all roles 
- wait for the cluster to be provisioned 

Results:
- the cluster becomes `active` although the node is in the `unavailable` state 
![image](https://user-images.githubusercontent.com/6218999/106330042-c3a89880-623f-11eb-8412-127c37953b1b.png)
![image](https://user-images.githubusercontent.com/6218999/106330102-e33fc100-623f-11eb-94ca-22066f7c0535.png)

- both the `launch kubectl` and `kubeconfig file` buttons are clickable 
- I can launch the kubectl on the UI 
- I can download the kubeconfig and use it to access the cluster
<img width="1451" alt="Screen Shot 2021-01-29 at 2 40 31 PM" src="https://user-images.githubusercontent.com/6218999/106330272-29952000-6240-11eb-8811-7caaec162576.png">

**Comment by jiaqiluo** [comment]
The bug is reproduced on rancher:`v2.5.5`

Steps:
- run rancher single-install
- add a custom cluster, edit the cluster as YAML and replace the default network settings with the following
```
network:
    plugin: none
```
- register one node with all roles
- wait for the cluster to be provisioned

Results:
-  the cluster is stuck in the `error` state
<img width="1429" alt="Screen Shot 2021-01-29 at 3 48 39 PM" src="https://user-images.githubusercontent.com/6218999/106335264-ae386c00-6249-11eb-9a9c-9e6de549f803.png">


- the `launch kubectl` and `kubeconfig file` are not clickable 
![image](https://user-images.githubusercontent.com/6218999/106335239-a4166d80-6249-11eb-8905-1c9e8f310ad4.png)

**Comment by jiaqiluo** [comment]
The following behavior is seen on 
- rancher/rancher:v2.5-head 2ae4c
- rancher/rancher:master-head d160b

Steps: the same as my previous comment

Results:
- the state of the cluster keeps switch between `updating` and `error` 
<img width="1014" alt="Screen Shot 2021-01-29 at 3 51 15 PM" src="https://user-images.githubusercontent.com/6218999/106336633-94e4ef00-624c-11eb-93c1-0132e58e3261.png">
<img width="1026" alt="Screen Shot 2021-01-29 at 3 51 06 PM" src="https://user-images.githubusercontent.com/6218999/106336638-96161c00-624c-11eb-9c3f-6c52382f4728.png">

- Both `launch kubectl` and `kubeconfig file` buttons are clickable, but only `kubeconfig file` works

![image](https://user-images.githubusercontent.com/6218999/106336672-a928ec00-624c-11eb-91df-6dc91f759680.png)

- error 1006 when launching kubectl
![image](https://user-images.githubusercontent.com/6218999/106336763-d6759a00-624c-11eb-907c-42c86e6452d4.png)

**Comment by superseb** [comment]
@jiaqiluo Creating a custom cluster with `none` as network plugin results for me in a cluster with the node in `NotReady` state but not in the `Error`/`Updating` state. Can you share the setup where this is reproduced so I can take a look? The only way I see the `Failed to Upgrade` happen is either when initial provisioning fails or when a second provisioning run is triggered (by editing cluster for instance).

Regarding `Launch kubectl`, my initial assessment is that when changing the way shell works this can no longer work. Initially this happened by shelling into a `cattle-node-agent` pod, which is always deployed because those DaemonSet pods tolerate all taints, and use HostNetwork so it wasnt depending on the CNI.

Now (as far as I know), we use steve in the `cattle-cluster-agent` to spawn a new pod for the shell. `cattle-cluster-agent` pod is not running because its a Deployment pod with limited tolerations. (this was changed due to wildcard tolerations causing rescheduling issues) and because the pod will wait for the CNI.

The kubeconfig works and can be used to complete the CNI installation. The only remaining issue is that kubectl in the UI does not work, to fix this we need to bypass steve and fallback to the `cattle-node-agent` as we used to do before as this is a running pod which we can access.

**Comment by superseb** [comment]
@jiaqiluo Can you confirm that the only difference today is that `Launch kubectl` doesn't work. (but kubeconfig works).

If this is confirmed to be the only difference, we need a decision on if we want to make `Launch kubectl` work which will be hard as no node will be ready to run a pod. You can still complete the CNI installation using the kubeconfig (which should work), so the question is specifically if that is good enough or if we want to make `Launch kubectl` work in this scenario.

**Comment by jiaqiluo** [comment]
Hi @superseb 
I can confirm that on the v2.5-head 5a5c039, `Launch Kubectl` doesn't work, but the kubeconfig file works (get it from the UI and use it locally on my mac). 

Also, I noticed that the cluster's state keeps as `Actvie` although the node is `Unavailable`. 

![image](https://user-images.githubusercontent.com/6218999/109853042-b271f380-7c12-11eb-8b22-371107a94d53.png)

<img width="1070" alt="Screen Shot 2021-03-03 at 11 22 50 AM" src="https://user-images.githubusercontent.com/6218999/109853141-d46b7600-7c12-11eb-8bf2-72b0418a62d9.png">

**Comment by deniseschannon** [comment]
With v2.5, this is expected behavior as we've moved the opening the shell into launching it's own pod. Without a CNI, there is no valid cluster that allows you to open the shell until a CNI is defined. We would expect the user to download the kube config and use that to apply other CNIs to the cluster.

This will not be fixed in v2.5 as this was a design choice going forward.

**Comment by sowmyav27** [comment]
Closing as per [these](https://github.com/rancher/rancher/issues/30427#issuecomment-797111290) comments
